### PR TITLE
PLUTO QA enhancements, expand `dcpy` repeat build functionality

### LINKED ISSUE #538: PLUTO QA enhancements
### GIS requests after QA of PLUTO 24v1
- Label value units where possible in graph tooltips and axes (% vs sq ft vs record count, etc.)
- Table of BBLs where bldgarea/lotarea > numfloors*2 aka “buildings containing unreasonably small apartments”
  - Hard to manually QA when there are ~400 records
  - One improvement would be if the report could identify new records added to this list based on the current build, hopefully it would be less daunting of a task if it was only 5-10 records each cycle.
  - Either filter to exclude or flag NYCHA BBLs

### LINKED ISSUE #657: PLUTO: Incorporate manual research
During the 24v1 creation records were identified where should be modified based on research:
1) 5008640515 building area value changed from 250 to 28500
Research top offenders from Unreasonably small apartments table and make modifications

### PR DESCRIPTION
**Pull request by fvankrieken**
Does NOT actually close #657, 

Closes #538

Part of #831 

Ended being a bunch of little stuff in here, so will go commit by commit

### QAQC Enhancements (#538) See issue for screenshots
1. Move logic to a component. Mainly wanted as part of commit 2 but this makes diffs in commit 2 easier to see
2. Add selectbox that lets you choose whether the other "diff" being shown is the previous, or previous of the same build type (major vs minor). The current build is a good example. We built 24v2. The prev is 24v1.1, so our "current" diff is "24v2 - 24v1.1". The former default, "previous previous" diff would be "24v1.1 - 24v1". But if we want to compare this to the last "major" diff instead, we can now do that, where the comparison diff will be "24v1 - 23v3.1" instead.
    There are cases where these would be the same - if we didn't have a minor last build and current was "24v2 - 24v1", either option would be "24v1 - 23v3.1". Same if we have 24v2.2 - both will be "24v2.1 - 24v2"
    This works for minors as well. 24v1.1 (with diff "24v1.1 - 24v1") can either look at diff "24v1 - 23v3.1" or "23v3.1 - 23v3"
    This commit includes a lot of renaming - instead of just "v1" "v2" and "v3" thought it's best to describe. So "version"/"v", "previous", "comparison", and "comparison_previous" seemed intuitive to me. "previous" always describing the directly preceding build, and "comparison" describing the build whose diff we want to compare to.
3. Adding new fields to outlier reports so that GIS can filter them out of the tables. So adding an additional column to the QA outputs as well as a checkbox in the app to filter rows. This checkbox is grayed out (with a useful tooltip) if the build doesn't have the new columns
4. add a few comments and axes labels as requested

### Non-correction or QA changes
5. don't vacuum in pluto build. This took a little while, and was added to attempt to reclaim space in constrained github runner temp postgres instances. Our db seems to have aggressive autovacuuming set up, so this seems unneeded
6. we've moved away from using .envs in this manner - I personally would rather run `set_env` manually before a build. Maybe worth broader discussion
7. Fix a bug introduced in #801 (or rather not entirely fixed - the fix was needed in 3 lines, I only added it on 2)
8. Rework "repeat" build functionality. I wanted to be able to repeat a draft, rather than only published versions, since once GIS begins QAing a dataset, it seems like we should be locking versions of all input datasets as to not introduce potentially new errors. This was tough with the way I initially implemented this. I decoupled normal recipe-planning and this repeat logic since they require fairly different inputs, and we can't have dynamic inputs to github actions

### CODE REVIEW

File: apps/qa/src/pages/pluto/components/outlier_report.py
Comment (fvankrieken) [inline apps/qa/src/pages/pluto/components/outlier_report.py:44]: Made these their own functions because of need for custom logic for filtering datasets

File: .github/workflows/build.yml
Comment (fvankrieken) [inline .github/workflows/build.yml:90]: env vars cannot be passed to workflow_calls (they're just not accessible at that part of the workflow), so they need to be here. But I thought it's a bit cleaner to have them defined up higher just to make them a bit clearer.

However, the "plan_command" had to be defined here because it depends on those vars

**Comment by fvankrieken** [comment]
Passing regular build [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9098664578)

Passing repeat build [here](https://github.com/NYCPlanning/data-engineering/actions/runs/9098664578)

File: dcpy/lifecycle/builds/plan.py
Comment (damonmcc) [inline dcpy/lifecycle/builds/plan.py:281]: (just a thought, not worth holding this up for)

this seems more like `version_type` since it's used to pick a type of `ProductKey`

**Review by damonmcc — APPROVED** [review]
LGTM! I like the `plan_command` concept

🤞🏾 for nightly QA after this is merged

File: dcpy/lifecycle/builds/plan.py
Comment (fvankrieken) [reply dcpy/lifecycle/builds/plan.py:281]: Yeah this was a tough one to name. Literally, more "product instance type" but that doesn't roll off the tongue or keyboard

**Comment by fvankrieken** [comment]
> LGTM! I like the `plan_command` concept
> 
> 🤞🏾 for nightly QA after this is merged

Nightly QA is passing already!

**Comment by fvankrieken** [comment]
Just rebased main, will make sure tests pass before merging

**Pull request merged**
